### PR TITLE
Changes the Console implementation for image upload

### PR DESCRIPTION
**Pull request by devantler**
- Moved from username/password authentication to Api Key auth, as username and password auth is not working.
- Added an option to select whether the image should be uploaded as a File or an Image Cropper.

Uncommented lines that make the code fail. DeliveryService is not working as expected right now. It fails with no meaningfull error.

### CODE REVIEW

File: samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs [hunk 1]
```diff
@@ -184,12 +181,26 @@ async Task UploadImageToMedia(ContentDeliveryService service, string projectAlia
...
+
+        media.SetValue("umbracoFile", rawPropertyValue, new FileInfoPart(new FileInfo(imagePath), fileName, $"image/{extension}"));
         var image = await managementService.Media.Create(media);
 
-        var newlyCreatedImage = await service.Media.GetById(image.Id);
-        RenderMediaWithUrl(newlyCreatedImage);
+        //var newlyCreatedImage = await contentDeliveryService.Media.GetById(image.Id);
+        //RenderMediaWithUrl(newlyCreatedImage);
     }
     else
```
[the remark this answers is not in the document]
Comment (devantler) [reply samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs:188]: Sure can; I pushed a change that removes the if-else and returns meaningful messages if either the `imagePath` or the `mediaName` is invalid.

File: samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs [hunk 2]
```diff
@@ -184,12 +181,26 @@ async Task UploadImageToMedia(ContentDeliveryService service, string projectAlia
...
+        var media = new Umbraco.Headless.Client.Net.Management.Models.Media { Name = mediaName, MediaTypeAlias = mediaTypeAlias, ParentId = folderId };
+
+        object? rawPropertyValue = isImageCropperUpload ? new { src = fileName } : isFileUpload ? fileName : null;
+
+        media.SetValue("umbracoFile", rawPropertyValue, new FileInfoPart(new FileInfo(imagePath), fileName, $"image/{extension}"));
         var image = await managementService.Media.Create(media);
 
-        var newlyCreatedImage = await service.Media.GetById(image.Id);
-        RenderMediaWithUrl(newlyCreatedImage);
+        //var newlyCreatedImage = await contentDeliveryService.Media.GetById(image.Id);
```
[the remark this answers is not in the document]
Comment (devantler) [reply samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs]: Yes and no. It works in the production environment but not in the development environment, so my initial thought was not to have it, as it did not work for all Umbraco projects. But it not working on the development environment is probably a non issue for most customers ;-)

File: samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs [hunk 3]
```diff
@@ -184,12 +181,26 @@ async Task UploadImageToMedia(ContentDeliveryService service, string projectAlia
...
+        var isFileUpload = false;
+        while (!isImageCropperUpload && !isFileUpload)
+        {
+            Console.WriteLine("Should the image be uploaded as an Image Cropper or a File? (image-cropper/file)");
+            var uploadType = Console.ReadLine()?.ToLower();
+            isImageCropperUpload = uploadType.Equals("image-cropper");
+            isFileUpload = uploadType.Equals("file");
+        }
+
+        var mediaTypeAlias = isImageCropperUpload ? "Image" : isFileUpload ? "File" : string.Empty;
```
[the remark this answers is not in the document]
Comment (devantler) [reply samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs]: You are correct. I did not like the implementation I did too much looking at it again, so I reworked it completely:

```
    var mediaTypeAlias = "";
    while (string.IsNullOrEmpty(mediaTypeAlias))
    {
        Console.WriteLine("Should the image be uploaded as an Image Cropper or a File? (image-cropper/file)");
        var uploadType = Console.ReadLine()?.ToLower();
        mediaTypeAlias = uploadType switch
        {
            "image-cropper" => "Image",
            "file" => "File",
            _ => ""
        };
    }

    var media = new Umbraco.Headless.Client.Net.Management.Models.Media { Name = mediaName, MediaTypeAlias = mediaTypeAlias, ParentId = folderId };
```

IMO this is more clear and keeps the operation on one variable.

File: samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs [hunk 4]
```diff
@@ -184,12 +181,26 @@ async Task UploadImageToMedia(ContentDeliveryService service, string projectAlia
...
+        var media = new Umbraco.Headless.Client.Net.Management.Models.Media { Name = mediaName, MediaTypeAlias = mediaTypeAlias, ParentId = folderId };
+
+        object? rawPropertyValue = isImageCropperUpload ? new { src = fileName } : isFileUpload ? fileName : null;
```
[the remark this answers is not in the document]
Comment (devantler) [reply samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs]: Fair I rewrote this one to an if-else.

File: samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs [hunk 5]
```diff
@@ -184,12 +181,26 @@ async Task UploadImageToMedia(ContentDeliveryService service, string projectAlia
...
 
-        var media = new Umbraco.Headless.Client.Net.Management.Models.Media {Name = mediaName, MediaTypeAlias = "Image", ParentId = folderId};
-        media.SetValue("umbracoFile", new { src = fileName }, new FileInfoPart(new FileInfo(imagePath), fileName, $"image/{extension}"));
+        var isImageCropperUpload = false;
+        var isFileUpload = false;
+        while (!isImageCropperUpload && !isFileUpload)
+        {
+            Console.WriteLine("Should the image be uploaded as an Image Cropper or a File? (image-cropper/file)");
+            var uploadType = Console.ReadLine()?.ToLower();
+            isImageCropperUpload = uploadType.Equals("image-cropper");
```
[the remark this answers is not in the document]
Comment (devantler) [reply samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs]: This is irrelevant, given my changes.

File: samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs [hunk 6]
```diff
@@ -256,7 +267,7 @@ async Task PageAndRenderDescendants(IContentDelivery contentDelivery, PagedConte
 
 void RenderContentWithUrl(Content content)
 {
-    Console.WriteLine("'"+ content.Name + "' on " + content.Url);
+    Console.WriteLine("'" + content.Name + "' on " + content.Url);
```
[the remark this answers is not in the document]
Comment (devantler) [reply samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs:279]: Yeah I deffo agree here :-) It is much more readable :-)


I prefer `Console.WriteLine($"'{content.Name}' on {content.Url}");` as it is a shorter syntax.

File: samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Fjellvang) [reply samples/Umbraco.Headless.Client.Samples.Console/Umbraco.Headless.Client.Samples.Console/Program.cs]: without knowing too much of this code, isn't it an issue that it doesn't work on dev or am i missing something?

I am atleast unfamiliar with code that doesn't work on dev but on prod?

**Pull request merged**